homeassistant/components/drop_connect/binary_sensor.py

```diff
@@ -0,0 +1,138 @@
+
+
+BINARY_SENSORS: list[DROPBinarySensorEntityDescription] = [
+    DROPBinarySensorEntityDescription(
+        key=LEAK_DETECTED,
+        translation_key=LEAK_DETECTED,
+        icon=LEAK_ICON,
+        device_class=BinarySensorDeviceClass.MOISTURE,
+        value_fn=lambda device: device.drop_api.leak_detected(),
+    ),
+    DROPBinarySensorEntityDescription(
+        key=PENDING_NOTIFICATION,
+        translation_key=PENDING_NOTIFICATION,
+        icon=NOTIFICATION_ICON,
+        value_fn=lambda device: device.drop_api.notification_pending(),
+    ),
+    DROPBinarySensorEntityDescription(
+        key=SALT_LOW,
+        translation_key=SALT_LOW,
+        icon=SALT_ICON,
+        value_fn=lambda device: device.drop_api.salt_low(),
+    ),
+    DROPBinarySensorEntityDescription(
+        key=RESERVE_IN_USE,
+        translation_key=RESERVE_IN_USE,
+        icon=WATER_ICON,
+        value_fn=lambda device: device.drop_api.reserve_in_use(),
+    ),
+    DROPBinarySensorEntityDescription(
+        key=PUMP_STATUS,
+        translation_key=PUMP_STATUS,
+        icon=PUMP_ICON,
+        value_fn=lambda device: device.drop_api.pump_status(),
+    ),
+]
+
+# Defines which binary sensors are used by each device type
+DEVICE_BINARY_SENSORS: dict[str, list[str]] = {
+    DEV_HUB: [LEAK_DETECTED, PENDING_NOTIFICATION],
+    DEV_LEAK_DETECTOR: [LEAK_DETECTED],
+    DEV_PROTECTION_VALVE: [LEAK_DETECTED],
+    DEV_PUMP_CONTROLLER: [LEAK_DETECTED, PUMP_STATUS],
+    DEV_RO_FILTER: [LEAK_DETECTED],
+    DEV_SALT_SENSOR: [SALT_LOW],
+    DEV_SOFTENER: [RESERVE_IN_USE],
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the DROP binary sensors from config entry."""
+    _LOGGER.debug(
+        "Set up binary sensor for device type %s with entry_id is %s",
+        config_entry.data[CONF_DEVICE_TYPE],
+        config_entry.entry_id,
+    )
+
+    if config_entry.data[CONF_DEVICE_TYPE] in DEVICE_BINARY_SENSORS:
+        async_add_entities(
+            DROPBinarySensor(hass.data[DOMAIN][config_entry.entry_id], sensor)
+            for sensor in BINARY_SENSORS
+            if sensor.key in DEVICE_BINARY_SENSORS[config_entry.data[CONF_DEVICE_TYPE]]
+        )
+
+
+class DROPBinarySensor(DROPEntity, BinarySensorEntity):
+    """Representation of a DROP binary sensor."""
+
+    entity_description: DROPBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DROPDeviceDataUpdateCoordinator,
+        entity_description: DROPBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize the binary sensor."""
+        super().__init__(entity_description.key, coordinator)
+        self.entity_description = entity_description
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the binary sensor."""
+        return self.entity_description.value_fn(self.coordinator) == 1
```

Comment:
What happens if a binary sensor is not available. If it is `-1`, should we return `None`, or are all sensors assumed to be available?

Reply:
All of the binary sensors will only be unavailable until the first payload is delivered.  The DROP hub will always deliver a 0 or 1 value. None of these binary sensors are optional.